contextual_orchestrator/cost_ledger.py

```
@@ -554,6 +554,54 @@ def __len__(self) -> int:
+_INSERT_DIMENSION_SQL = {
+    "qmark": (
+        "INSERT INTO cost_attribution_dimensions "
+        "(dimension_name, dimension_label, dimension_order) VALUES (?, ?, ?)"
+    ),
+    "pyformat": (
+        "INSERT INTO cost_attribution_dimensions "
+        "(dimension_name, dimension_label, dimension_order) VALUES (%s, %s, %s)"
+    ),
+}
+_INSERT_USAGE_SQL = {
+    "qmark": f"INSERT INTO llm_usage_records ({_USAGE_COLUMN_SQL}) VALUES ({_QMARK_USAGE_VALUES})",
```

Ruff (0.16.1)

[error] Possible SQL injection vector through string-based query construction

(S608)